src/main/java/com/dynamic/Fibonacci.java

```diff
@@ -0,0 +1,33 @@
+package src.main.java.com.dynamic;
+
+import java.util.HashMap;
+
+/**
+ * The Fibonacci sequence is a sequence of numbers in which each number after the first two is the sum of the two preceding ones.
+ * 1, 1, 2, 3, 5, 8, 13, ...
+ */
+public class Fibonacci {
+
+    private static HashMap<Integer, Integer> fibonacciCache = new HashMap<>();
+
+    /**
+     * A recursive method that calculates and returns the nth fibonacci number.
+     *
+     * @return the nth fibonacci number.
+     */
+    public int fibonacci(int n) {
```

Comment:
Instead of a recursive approach or top-down dynamic approach, A bottom-up approach will do the trick.